src/handlers/secret.rs

```diff
@@ -0,0 +1,182 @@
+use log::debug;
+use rocket::http::Status;
+use rocket::response::status;
+use rocket::serde::json::Json;
+use rocket::{Response, State};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use serde_json::Value;
+use uuid::Uuid;
+use crate::rocket::futures::TryStreamExt;
+
+use crate::Storage;
+use crate::{
+    error::AppError,
+    storage::{
+        secret::{NewSecret, SecretEntity},
+        storage::AttemptCountRule,
+    },
+};
+
+#[derive(Serialize)]
+pub struct AllSecretsResponse {
+    secrets: Vec<Secret>,
+}
+
+#[derive(Clone, Deserialize)]
+pub struct GuessSecret {
+    guess: String,
+    username: String,
+}
+
+#[derive(Clone, Debug, Serialize)]
+pub struct Secret {
+    pub id: Uuid,
+    pub guessed: bool,
+    pub clue1: Option<String>,
+    pub clue2: Option<String>,
+    pub clue3: Option<String>,
+    pub guesser: Option<String>,
+    pub secret: Option<String>,
+}
+
+pub fn process_secret(attempt_rule: AttemptCountRule, entity: SecretEntity) -> Secret {
+    Secret {
+        id: entity.id,
+        guessed: entity.guesser.is_some(),
+        clue1: match entity.guess_attempts >= attempt_rule.clue1_attempts {
+            true => Some(entity.clue1),
+            false => None,
+        },
+        clue2: match entity.guess_attempts >= attempt_rule.clue2_attempts {
+            true => Some(entity.clue2),
+            false => None,
+        },
+        clue3: match entity.guess_attempts >= attempt_rule.clue3_attempts {
+            true => Some(entity.clue3),
+            false => None,
+        },
+        secret: entity.guessed_secret,
+        guesser: entity.guesser,
+    }
+}
+
+#[post("/secrets", format = "json", data = "<secret>")]
+pub async fn create_secret_handler(
+    storage: &State<Storage>,
+    attempt_rule: &State<AttemptCountRule>,
+    secret: Json<NewSecret>,
+) -> Result<Json<Secret>, status::Custom<Json<Value>>> {
+    let created_secret = storage
+        .create_secret(secret.into_inner())
+        .await;
+
+    let secret = match created_secret {
+        Ok(secret) => secret,
+        Err(_) => return Err(status::Custom(Status::InternalServerError, Json(json!({"error": "Internal Server Error."}))))
+    };
+
+    let processed_secret = process_secret(*attempt_rule.inner(), secret);
+
+    debug!("Create Secret Handler executed successfully.");
```

Comment:
What you say we add the processed secret to this debug log?

So in the future, when we're debugging the service, we can check which request succeeded and which failed based on what it returned

Reply:
```
debug!("Get Secret Handler executed successfully and returned a {:?}.", processed_secret);
```